### DISCUSSION TITLE
working of yolov5

### DISCUSSION THREAD

**Discussion by jb-ssncse** [ultralytics/yolov5#2192] [Q&A]
Hello,
Can you help me with  understanding the architecture of yolov5

**Comment by glenn-jocher** [comment]
@jb-ssncse the model architecture is defined in the model yaml files. For example here is YOLOv5s:
https://github.com/ultralytics/yolov5/blob/master/models/yolov5s.yaml

**Reply by jb-ssncse** [reply]
Thank you, In that case do  i need to reset nc to my dataset classes in that .yaml?

**Reply by glenn-jocher** [reply]
If your model yaml/weights and data yaml differ in class counts then YOLOv5 will always use your dataset class counts, so there's nothing to worry about.
